src/test/java/org/klodnicki/service/ReturnBookServiceTest.java

```diff
@@ -0,0 +1,191 @@
+import org.klodnicki.repository.BookRepository;
+import org.klodnicki.repository.LendingInformationRepository;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ReturnBookServiceTest {
+
+    private final BookRepository bookRepository = new BookRepository();
+    private final ReturnBookService returnBookService = new ReturnBookService(new AccountService(), new BookService(),
+            new LendingInformationService());
+
+    private final AccountRepository accountRepository = new AccountRepository();
+    private final LendingInformationRepository lendingInformationRepository = new LendingInformationRepository();
+    private final BookService bookService = new BookService();
+
+    @BeforeEach
+    public void cleanDatabase() {
+        bookRepository.removeAll();
+        accountRepository.removeAll();
+        lendingInformationRepository.removeAll();
+    }
+
+    @AfterEach
+    public void removeForeignKeyFromDatabase() {
+        lendingInformationRepository.removeAll();
+    }
+
+    @Test
+    public void prepareListOfBorrowedBooksByAccount_ShouldThrowNotFoundInDatabaseException_WhenAccountIsNotFoundInDatabase() {
+        //given
+        String firstName = "";
+        String lastName = "";
+        String pesel = "";
+
+        //when
+        assertThrows(NotFoundInDatabaseException.class, () -> returnBookService.prepareListOfBorrowedBooksByAccount
+                (firstName, lastName, pesel));
+    }
+
+    @Test
+    public void prepareListOfBorrowedBooksByAccount_ShouldReturnListOfStrings_WhenAccountHasBorrowedBooks()
+            throws NotFoundInDatabaseException {
+        //given
+        BookInfo bookInfo = new BookInfo("title", "author", "isbn", "publisher",
+                2011, "edition", "genre", "description", "language", 100);
+        BookInfo bookInfo2 = new BookInfo("title2", "author2", "isbn2", "publisher2",
+                2000, "edition2", "genre2", "description2", "language2", 1000);
+
+        bookRepository.add(bookInfo);
+        bookRepository.add(bookInfo2);
+
+        Account account = new Account("first name", "second name", "last name", "pesel",
+                "phone number", "email", "address");
+
+        accountRepository.create(account);
+
+        LendingInformation lendingInformation1 = new LendingInformation(bookInfo, account, LocalDateTime.now());
+        LendingInformation lendingInformation2 = new LendingInformation(bookInfo2, account, LocalDateTime.now());
+
+        lendingInformationRepository.add(lendingInformation1);
+        lendingInformationRepository.add(lendingInformation2);
+
+        List<BookInfo> records = new ArrayList<>();
+        records.add(bookInfo);
+        records.add(bookInfo2);
+
+        List<String> expected = new ArrayList<>();
+        for (BookInfo record : records) {
+            expected.add(record.toString());
+        }
```

Comment:
Those lines are making code less readable. Make it in one List, added directly to `expected` instead of from one list to another, please. Will look much more readable.

Reply:
I don't know exactly what you mean, but I used stream(). Hopefully that works fine for you too!

Reply:
![image](https://github.com/BartoliniAndBorderCollies/Library_Manager_Official/assets/126821059/cae2ce17-5430-4798-b90a-5e21444ee116)